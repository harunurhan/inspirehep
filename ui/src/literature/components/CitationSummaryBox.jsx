import React from 'react';
import { Col, Row, Checkbox } from 'antd';
import PropTypes from 'prop-types';

import ContentBox from '../../common/components/ContentBox';
import CitationSummaryGraphContainer from '../../common/containers/CitationSummaryGraphContainer';
import CitationSummaryTableContainer from '../../common/containers/CitationSummaryTableContainer';
import AuthorizedContainer from '../../common/containers/AuthorizedContainer';
import { SUPERUSER_OR_CATALOGER } from '../../common/authorization';

function CitationSummaryBox({
  namespace,
  excludeSelfCitations,
  onExcludeSelfCitationsChange,
}) {
  return (
    <ContentBox subTitle="Citation Summary">
      <AuthorizedContainer authorizedRoles={SUPERUSER_OR_CATALOGER}>
        <Checkbox
          onChange={event => onExcludeSelfCitationsChange(event.target.checked)}
          checked={excludeSelfCitations}
        >
          Exclude self-citations
        </Checkbox>
      </AuthorizedContainer>
      <Row gutter={{ xs: 0, lg: 32 }}>
        <Col span={24}>
          <CitationSummaryTableContainer />
        </Col>
        <Col span={24}>
          <CitationSummaryGraphContainer namespace={namespace} />
        </Col>
      </Row>
    </ContentBox>
  );
}

CitationSummaryBox.propTypes = {
  namespace: PropTypes.string.isRequired,
  excludeSelfCitations: PropTypes.bool.isRequired,
  onExcludeSelfCitationsChange: PropTypes.func.isRequired,
};

export default CitationSummaryBox;
