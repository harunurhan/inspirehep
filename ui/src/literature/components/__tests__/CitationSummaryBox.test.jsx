import React from 'react';
import { shallow } from 'enzyme';

import CitationSummaryBox from '../CitationSummaryBox';

describe('CitationSummaryBox', () => {
  it('renders', () => {
    const wrapper = shallow(
      <CitationSummaryBox query={{ q: 'cern' }} onQueryChange={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  // TODO: enable after https://github.com/airbnb/enzyme/issues/2086
  xit('calls onQueryChange initially and when query is different', () => {
    const onQueryChange = jest.fn();
    const initialQuery = { q: 'cern' };
    const wrapper = shallow(
      <CitationSummaryBox query={initialQuery} onQueryChange={onQueryChange} />
    );
    const newQuery = { experiment: 'cms' };

    wrapper.setProps({ query: newQuery });

    expect(onQueryChange).toHaveBeenCalledWith(initialQuery);
    expect(onQueryChange).toHaveBeenCalledWith(newQuery);
  });
});
