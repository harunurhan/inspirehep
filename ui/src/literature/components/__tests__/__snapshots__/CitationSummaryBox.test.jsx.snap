// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitationSummaryBox renders 1`] = `
<ContentBox
  className=""
  leftActions={null}
  loading={false}
  rightActions={null}
  subTitle="Citation Summary"
  title={null}
>
  <Row
    gutter={
      Object {
        "lg": 32,
        "xs": 0,
      }
    }
  >
    <Col
      lg={7}
      xs={24}
    >
      <Connect(CitationSummaryTable) />
    </Col>
    <Col
      lg={17}
      xs={24}
    >
      <Connect(toAllJS(CitationSummaryGraph))
        namespace="literature"
      />
    </Col>
  </Row>
</ContentBox>
`;
